### ISSUE TITLE
Fatal error using try! 

### ISSUE THREAD

**Issue by Silve2611** [stephencelis/SQLite.swift#430]
As already mentioned in #339 the try! will cause a program to crash if multiple programs access the database at the same time. 
This is not only happening when using Scalar but also when using 

> SELECT \* FROM table

```
let table = Table("TProperties")
var all = [Row]()
do{
    all = Array(try db.prepare(table))
}catch{
     print("getSettings failed")
}
```

The reason behind it is the Generator type 

```
extension Statement : GeneratorType {
    public func next() -> [Binding?]? {
        return try! step() ? Array(row) : nil
    }
}
```

My solution for it is the following change but i couldn't test yet if it is really working 

```
extension Statement : GeneratorType {
    public func next() -> [Binding?]? {
        var result:[Binding?]?
        do{
            result = try step() ? Array(row) : nil
        }catch{
            result = nil
        }
        return result
        //return try! step() ? Array(row) : nil
    }
}
```

**Comment by blkbam** [comment]
I'm seeing the same thing occur. I'm crashing while iterating through the results. @Silve2611 did this change work for you?

**Comment by nickmshelley** [comment]
I'm hoping https://github.com/stephencelis/SQLite.swift/pull/413 will get merged at some point, which I think should take care of this. Unfortunately it's dependent on https://github.com/stephencelis/SQLite.swift/pull/451, but once that's through I'll try to get the error handling one to a good place.

**Comment by blkbam** [comment]
Just the change above alone to next() seems to do it in my short time testing however I'm sure the other changes are beneficial. I'll go through some more testing and take a look at the other PR's to see if I need to branch or hold off.

**Comment by Silve2611** [comment]
Yes you are right. I had to modify the next() method to solve this. I helped myself by rewriting the statement function 

`extension Statement : GeneratorType {

    public func next() -> [Binding?]? {
        var result:[Binding?]?
        do{
            result = try step() ? Array(row) : nil
        }catch{
            result = nil
        }
        return result
        //this was the old code
        //return try! step() ? Array(row) : nil 
    }
}`

**Comment by jberkel** [comment]
fixed in 5a9b583d2a28d12bc8523b204e75a0da6c8b04ff
